spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClient.java

```diff
@@ -0,0 +1,681 @@
+ * acceptingCommits: Boolean --- Whether the commit owner is accepting new commits. This will only
+ *  be set to false when the table is converted from managed commits to file system commits.
+ * tableVersion: Number --- The version of the latest commit.
+ * tableTimestamp: Number --- The inCommitTimestamp of the latest commit.
+ * schemaVersion: Number --- The version of the schema used to store the data.
+ * commits: --- The list of unbackfilled commits.
+ *  version: Number --- The version of the commit.
+ *  inCommitTimestamp: Number --- The inCommitTimestamp of the commit.
+ *  fsName: String --- The name of the unbackfilled file.
+ *  fsLength: Number --- The length of the unbackfilled file.
+ *  fsTimestamp: Number --- The modification time of the unbackfilled file.
+ */
+public class DynamoDBCommitOwnerClient implements CommitOwnerClient {
+
+    /**
+     * The name of the DynamoDB table used to store unbackfilled commits.
+     */
+    final String managedCommitsTableName;
+
+    /**
+     * The DynamoDB client used to interact with the DynamoDB table.
+     */
+    final AmazonDynamoDB client;
+
+    /**
+     * The endpoint of the DynamoDB table.
+     */
+    final String endpoint;
+
+    /**
+     * The number of write capacity units to provision for the DynamoDB table if the
+     * client ends up creating a new one.
+     */
+    final long writeCapacityUnits;
+
+    /**
+     * The number of read capacity units to provision for the DynamoDB table if the
+     * client ends up creating a new one.
+     */
+    final long readCapacityUnits;
+
+    /**
+     * The number of commits to batch backfill at once. A backfill is performed
+     * whenever commitVersion % batchSize == 0.
+     */
+    public final long backfillBatchSize;
+
+    /**
+     * Whether we should skip matching the current table path against the one stored in DynamoDB
+     * when interacting with it.
+     */
+    final boolean skipPathCheck;
+
+    /**
+     * The key used to store the tableId in the managed commit table configuration.
+     */
+    final static String TABLE_CONF_TABLE_ID_KEY = "tableId";
+
+    /**
+     * The version of the client. This is used to ensure that the client is compatible with the
+     * schema of the data stored in the DynamoDB table. A client should only be able to
+     * access a table if the schema version of the table matches the client version.
+     */
+    final int CLIENT_VERSION = 1;
+
+
+    public DynamoDBCommitOwnerClient(
+            String managedCommitsTableName,
+            String endpoint,
+            AmazonDynamoDB client,
+            long backfillBatchSize) throws IOException {
+        this(
+            managedCommitsTableName,
+            endpoint,
+            client,
+            backfillBatchSize,
+            5 /* readCapacityUnits */,
+            5 /* writeCapacityUnits */,
+            false /* skipPathCheck */);
+        tryEnsureTableExists();
```

Comment:
We are calling the other constructor here so it will take care of invoking tryEnsureTableExists ?

Reply:
You are right, will fix